### ISSUE TITLE
Autodoc does not work with enum subclasses with properties/methods

### ISSUE THREAD

**Issue by albireox** [sphinx-doc/sphinx#5436]
Subject: Autodoc does not work with enum subclasses with properties/methods

### Problem
- When a subclass of `enum.Flag` or `enum.IntFlag` has additional methods or properties, those are not displayed by autodoc.

#### Procedure to reproduce the problem

I have this enumeration 

```python
class Maskbit(enum.Flag):
    """A maskbit enumeration. Intended for subclassing."""

    @property
    def active_bits(self):
        """Returns a list of flags that match the value."""

        return [bit for bit in self.__class__ if bit.value & self.value]
```

When documenting it with

```rst
.. automodule:: jaeger.utils.maskbits
    :undoc-members:
```

it shoes the doctoring for the class but not the property or its docstring. The same is true if `active_bits` is a method.

#### Error logs / results
- No errors but the output is not the expected one.

#### Expected results
- Subclasses of enum should be documented as normal classes.

### Reproducible project / your project
- See snippet above. If needed I can provide the full project but shouldn't be necessary.

### Environment info
- OS: macOS 10.14
- Python version: 3.7.0
- Sphinx version: v1.8.0

**Comment by tk0miya** [comment]
Thank you for reporting!

Note:
With 3.7, it seems `active_bits` is not listed on the attributes list:
```
>>> dir(Maskbit)
['__class__', '__doc__', '__members__', '__module__']
>>> Maskbit.__members__
mappingproxy(OrderedDict())
```

And I found it on internal dictionary:  `__dict__`
```
>>> Maskbit.__dict__
mappingproxy({'_generate_next_value_': <function Flag._generate_next_value_ at 0x10405e2f0>, '__module__': '__main__', '__doc__': 'A maskbit enumeration. Intended for subclassing.', 'active_bits': <property object at 0x103f85098>, '_member_names_': [], '_member_map_': OrderedDict(), '_member_type_': <class 'object'>, '_value2member_map_': {}, '__new__': <function Enum.__new__ at 0x10405da60>})
```

I don't know other interpreters works as same.

**Comment by albireox** [comment]
Seems the same is true in 3.6.

**Comment by tk0miya** [comment]
Fixed by #5449.
Thanks,
